crux/src/Crux/UI/JS.hs

```diff
@@ -3,33 +3,61 @@
 -- | Utilites for generating JSON
 module Crux.UI.JS where
 
-import Data.Text(unpack)
+import Data.Text(unpack, Text)
+import qualified Data.Text as Text
+import Numeric
 import Data.List(intercalate)
 import Data.Maybe(fromMaybe)
 import System.Directory( canonicalizePath )
 
+import Data.Aeson.Encode.Pretty (encodePretty)
+import Data.Aeson (decode, Value)
+import qualified Data.ByteString.Lazy.Char8 as BL
+
+
 import What4.ProgramLoc
 
+-- |`jsLoc` takes a program location, and optionally returns it
+-- as `JS String`
 jsLoc :: ProgramLoc -> IO (Maybe JS)
 jsLoc x =
   case plSourceLoc x of
-    SourcePos f l c ->
-      do let fstr = unpack f
-         fabsolute <-
-            if | null fstr -> pure ""
-               | otherwise -> canonicalizePath fstr
-         pure $ Just $ jsObj
-           [ "file" ~> jsStr fabsolute
-           , "line" ~> jsStr (show l)
-           , "col"  ~> jsStr (show c)
-           ]
+    SourcePos fname l c -> parsePos fname l c
+    -- |Attempt to parse `OtherPos` in case it is in fact a code span:
```

Comment:
No need to use Haddock formatting here, since that only works for top-level functions anyway:

```suggestion
    -- Attempt to parse `OtherPos` in case it is in fact a code span:
```